chemprop/featurizers/molecule.py

```diff
@@ -41,3 +43,28 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        warnings.warn(
+            "The RDKit 2D features can deviate signifcantly from a normal distribution. Consider "
+            "manually scaling them using an appropriate scaler before creating datapoints, rather "
+            "than using the scikit-learn `StandardScaler` (the default in Chemprop)."
+        )
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        if mol.GetNumHeavyAtoms() == 0:
+            features = np.array(
+                [0 if name == "SPS" else func(mol) for name, func in Descriptors.descList], float
+            )
+        else:
+            features = np.array([func(mol) for name, func in Descriptors.descList], float)
+
+        NAN_TOKEN = 0
```

Comment:
Not sure I see the value in setting this here and then using it immediately versus just adding a comment that says 'set np.nan to zero'

Reply:
The main value is that it is consistent with how we deal with nans in the extra features/descriptors the user gives when making a data point (https://github.com/chemprop/chemprop/blob/06171ace741b1641deddf978a96d99a68236c352/chemprop/data/datapoints.py#L35). That said, we already take care of this in datapoint, so probably I can remove this nan_to_zero conversion and return an array that potentially has some nans. This is different than what I previously thought, that the featurizers were responsible for not returning nans. Currently, I'm of the opinion we shouldn't hide the nans from the user, except when we need to like when making the datapoints because the model can't take nans.